### ISSUE TITLE
Switch to using Persistent Volumes

### ISSUE THREAD

**Issue by jwhitcraft** [drud/vault-consul-on-kube#14]
Right now, this is setup to use gcloud, but if it's switched to using Persistent Volumes and Persistent Volumes Claims in kubernetes, this can easily also be supported on AWS.

**Comment by rfay** [comment]
I don't think we will end up doing this, but patches are welcome. I was hoping to switch to using stateful sets instead, which would also work on AWS I think, per the latest version of https://github.com/kelseyhightower/consul-on-kubernetes

Our actual production needs (the reason we built this) are focused on GKE, but it would in fact be preferable if it were neutral about the back-end architecture. But since we're using it in production on GKE, that will be our support focus.

**Comment by jwhitcraft** [comment]
@rfay i'm gonna submit a PR for this in just a sec, to me it's a stop gap till i can get the Stateful sets running when i upgrade Consul which will undo all of this but i'm not sure when i'll fully get that running 100% so I figure this will work for now.

**Comment by sworisbreathing** [comment]
@jwhitcraft I tried stateful sets yesterday after seeing them in https://github.com/kelseyhightower/consul-on-kubernetes. I had a lot of difficulty with the persistent volumes in the stateful set using local disk (`hostPath`) and eventually gave up on it. But I admit I don't really understand how PVs work in kubernetes so it might just be something I did wrong in my local setup.

**Comment by jwhitcraft** [comment]
@sworisbreathing, I never tried it locally on minikube, I've just always run it in my test cluster.
